### PR TITLE
fix(responsive): improve responsive design for various dashboard components

### PR DESCRIPTION
**Pull request by gregfromstl**
### TL;DR

Improved the layout and accessibility of various dashboard components to enhance user experience.

### What changed?

- Updated `sidebar.client.tsx` to adjust width classes for different screen sizes.
- Revised `add-partner-form.client.tsx` for a more responsive design, changed from space-x-* to flex-column on mobile and added necessary gap and hidden classes.
- Modified `ecosystem-header.client.tsx` to optimize layout for better readability and accessibility with skeleton loaders and adjusted button visibility based on screen size.
- Enhanced `integration-permissions-section.client.tsx` to better support various screen sizes, changed layout for radio buttons to stack on smaller screens.
- Refined `partners-table.tsx` to enhance table readability on smaller screens with new class adjustments.
- Adjusted `layout.tsx` to better handle padding on different screen sizes.
- Updated `create-ecosystem-form.client.tsx` and related pages to improve the form layout on smaller screens.
- Included Tooltip component for better accessibility in `ecosystem.page.tsx` and wrapped `AppRouterProviders` with `TooltipProvider` to enable tooltips across the app.

### How to test?

1. Navigate through the dashboard pages including sidebar, add partner form, ecosystem header, integration permissions, partners table, and creating ecosystems.
2. Verify that the layout adjustments enhance the user interface on different screen sizes (mobile, tablet, desktop).
3. Check that all tooltips appear and display relevant information.

### Why make this change?

To improve the user experience by making the dashboard more responsive, accessible, and visually appealing across different devices and screen sizes.

---

 

---

## PR-Codex overview
The focus of this PR is to update the UI components and layouts in the dashboard ecosystem management feature.

### Detailed summary
- Updated styling and layout of various components in the ecosystem management feature
- Added form fields and buttons for creating and managing ecosystem partners
- Improved responsiveness and accessibility of UI elements for better user experience

> The following files were skipped due to too many changes: `apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/page.tsx`, `apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/[ecosystemId]/components/client/ecosystem-header.client.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by gregfromstl** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3557?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3557** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3557?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#3556** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3556?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3537** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3537?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#3535](https://github.com/thirdweb-dev/js/pull/3535) <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3535?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gregfromstl and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/[ecosystemId]/components/client/ecosystem-header.client.tsx [hunk 1]
```diff
@@ -30,39 +30,46 @@ export function EcosystemHeader({ ecosystemId }: { ecosystemId: string }) {
...
-            {currentEcosystem.name}
-          </h2>
-        )}
-        {!currentEcosystem ? (
-          <Skeleton className="h-5 rounded-md w-[300px]" />
-        ) : (
-          <p className="text-muted-foreground">{currentEcosystem.slug}</p>
-        )}
+    <header>
+      <div className="grid space-y-4 md:space-y-0 md:grid-cols-4">
```
Comment (jnsdls) [inline apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/[ecosystemId]/components/client/ecosystem-header.client.tsx:34]: ~space~ -> gap

File: apps/dashboard/src/app/(dashboard)/providers.tsx [hunk 2]
```diff
@@ -13,11 +14,13 @@ export function AppRouterProviders(props: { children: React.ReactNode }) {
     <QueryClientProvider client={queryClient}>
       <AllChainsProvider>
         <ChainsProvider>
-          <DashboardThirdwebProviderSetup>
-            <ThemeProvider attribute="class" defaultTheme="dark">
-              {props.children}
-            </ThemeProvider>
-          </DashboardThirdwebProviderSetup>
+          <TooltipProvider>
```
Comment (jnsdls) [inline apps/dashboard/src/app/(dashboard)/providers.tsx:17]: oof, do we truly need this? why didn't we need it before when we already have tooltips in chainlist etc for example?

**Review by jnsdls — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/app/(dashboard)/providers.tsx (hunk 2 above)
Comment (gregfromstl) [reply apps/dashboard/src/app/(dashboard)/providers.tsx:17]: Oh looks like you made a nice little component that does it all for you, will use that

**Pull request merged**